### PR TITLE
Auxiliar branch for 2nd task Connect Data to Api

### PR DESCRIPTION
**Pull request by nicosanchezprev**
**PR for 2nd task Connect Data to Api**

- I made this branch to make the corrections for 2nd task.

### CODE REVIEW

File: components/ItemCripto/index.tsx [hunk 1]
```diff
@@ -8,31 +8,56 @@ import {
...
+    return q.match(regex)[0];
+  } else {
+    let t = n.toString();
+    let regex = /(\d*.\d{0,2})/;
+    return t.match(regex)[0];
+  }
+};
+
+const ItemCripto = ({item}: ItemCriptoProps) => {
+  return (
```
Comment (tomasdematos) [inline components/ItemCripto/index.tsx]: implicit return

File: components/ItemCripto/styles.ts [hunk 2]
```diff
@@ -32,7 +37,18 @@ export const CriptoPercentage = styled.Text<CriptoPercentageProps>`
...
   text-align: right;
 `;
+export const CriptoPrice = styled.Text`
+  text-align: right;
+  font-weight: 600;
+  font-size: 16px;
+  color: ${theme.colors.black};
+`;
+
+export const ImageArrow = styled.Image``;
```
Comment (tomasdematos) [inline components/ItemCripto/styles.ts]: empty style

File: redux/reducersComp/cryptosSlice.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const initialState: initialStateProps = {
+  cryptosData: [],
+  error: '',
+};
+
+export const cryptoApiData = createAsyncThunk(
+  'crypto/api',
+  async (name: string) => {
+    const response = await axios.get(
+      `https://data.messari.io/api/v1/assets/${name}/metrics`,
```
Comment (tomasdematos) [inline redux/reducersComp/cryptosSlice.ts]: keep this: https://data.messari.io/api/v1/
as the base url, stored in a .env file. 
https://yarnpkg.com/package/react-native-dotenv

File: redux/reducersComp/cryptosSlice.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+const cryptosSlice = createSlice({
+  name: 'cryptos',
+  initialState,
+  reducers: {
+    addCrypto: (state, action: PayloadAction<DataCriptoInfo | null>) => {
+      if (action.payload === null) {
+        return;
+      }
+      let check = state.cryptosData.find(
```
Comment (tomasdematos) [inline redux/reducersComp/cryptosSlice.ts]: use const

File: redux/reducersComp/cryptosSlice.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      if (check === undefined) {
+        state.cryptosData.push(action.payload);
+      } else {
+        return;
+      }
+    },
+    cleanError: state => {
+      state.error = '';
+    },
+    deleteCrypto: (state, action: PayloadAction<string>) => {
```
Comment (tomasdematos) [inline redux/reducersComp/cryptosSlice.ts]: this is out of scope, is from the card number 4

File: redux/reducersComp/cryptosSlice.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      state.cryptosData = state.cryptosData.filter(
+        crypto => crypto.name !== action.payload,
+      );
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(cryptoApiData.fulfilled, (state, action) => {
+      let check = state.cryptosData.find(
```
Comment (tomasdematos) [inline redux/reducersComp/cryptosSlice.ts]: use const

File: redux/reducersComp/cryptosSlice.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      return response.data.status;
+    }
+  },
+);
+
+const cryptosSlice = createSlice({
+  name: 'cryptos',
+  initialState,
+  reducers: {
+    addCrypto: (state, action: PayloadAction<DataCriptoInfo | null>) => {
```
Comment (tomasdematos) [inline redux/reducersComp/cryptosSlice.ts]: are you using this reducer?

File: components/ItemCripto/index.tsx [hunk 8]
```diff
@@ -8,31 +8,56 @@ import {
...
-              ? require('../../assets/img/greenarrow.png')
-              : require('../../assets/img/redarrow.png')
-          }
-        />
-        {Math.abs(item.percentage) + '%'}
-      </CriptoPercentage>
-    </View>
-  </CriptoContainer>
-);
+export const truncPrice = (n: any) => {
```
Comment (FedericoLeiva12) [inline components/ItemCripto/index.tsx]: It should be done using toFixed, not using RegExp

File: redux/reducersComp/cryptosSlice.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      if (action.payload === null) {
+        return;
+      }
+      let check = state.cryptosData.find(
+        elem => elem.name === action.payload?.name,
+      );
+      if (check === undefined) {
+        state.cryptosData.push(action.payload);
+      } else {
+        return;
```
Comment (FedericoLeiva12) [inline redux/reducersComp/cryptosSlice.ts]: Is this return necessary?

File: components/ItemCripto/index.tsx [hunk 10]
```diff
@@ -8,31 +8,56 @@ import {
...
+      <CriptoContainer>
+        <SeconContainer>
+          <CriptoImg source={item.img} />
+          <View>
+            <CriptoName>{item.name}</CriptoName>
+            <Text>{item.symbol}</Text>
+          </View>
+        </SeconContainer>
+        <View>
+          <CriptoPrice>{'$' + truncPrice(item.price)}</CriptoPrice>
```
Comment (VicenBayugar) [inline components/ItemCripto/index.tsx]: use item.price.toFixed(2) and delete truncPrice function

File: redux/reducersComp/cryptosSlice.ts (hunk 7 above)
Comment (nicosanchezprev) [reply redux/reducersComp/cryptosSlice.ts]: Im using it but in card 4, i will deleted it for this card.

File: components/ItemCripto/index.tsx (hunk 10 above)
Comment (nicosanchezprev) [reply components/ItemCripto/index.tsx]: Wow, i wish i new this function before haha. I will change that!

File: components/ItemCripto/index.tsx (hunk 8 above)
Comment (nicosanchezprev) [reply components/ItemCripto/index.tsx]: I already fix it with the correction from Vicen in the previous comment!

Comment (VicenBayugar) [reply components/ItemCripto/index.tsx]: delete the function if you're not going to use it

File: redux/reducersComp/cryptosSlice.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+const initialState: initialStateProps = {
+  cryptosData: [],
+  error: '',
+};
+
+export const cryptoApiData = createAsyncThunk(
+  'crypto/api',
+  async (name: string) => {
+    console.log(API_URL, ICONS_URL);
```
Comment (VicenBayugar) [inline redux/reducersComp/cryptosSlice.ts]: delete this console.log

File: components/ItemCripto/index.tsx (hunk 8 above)
Comment (FedericoLeiva12) [reply components/ItemCripto/index.tsx]: Delete it if you're not going to use.

File: components/ListCripto/index.tsx [hunk 12]
```diff
@@ -1,54 +1,32 @@
...
-    id: '3',
-    name: 'XRP',
-    symbol: 'XRP',
-    price: '$0.220568',
-    percentage: -2.47,
-    img: require('../../assets/img/xrp.png'),
-  },
-];
+const ListCripto = ({setModal}: ListCriptoProps) => {
+  const cryptosState = useAppSelector(state => state.crypto.cryptosData);
```
Comment (FedericoLeiva12) [inline components/ListCripto/index.tsx]: use destructuring
({ crypto }) or ({ crypto: {cryptosData} })

File: redux/reducersComp/cryptosSlice.ts [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(cryptoApiData.fulfilled, (state, action) => {
+      const check = state.cryptosData.find(
+        elem => elem.name === action.payload.name,
+      );
+      if (check === undefined) {
+        state.cryptosData.push(action.payload);
+      } else {
+        state.error = 'This cryptocurrencie is already added!';
```
Comment (FedericoLeiva12) [inline redux/reducersComp/cryptosSlice.ts]: Cryptocurrency*

File: redux/reducersComp/cryptosSlice.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    });
+    builder.addCase(cryptoApiData.rejected, state => {
+      state.error = 'This cryptocurrencie do not exist!';
```
Comment (FedericoLeiva12) [inline redux/reducersComp/cryptosSlice.ts]: Cryptocurrency*

File: redux/reducersComp/cryptosSlice.ts [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    });
+  },
+});
+
+export const selectCount = (state: RootState) => state.crypto;
```
Comment (FedericoLeiva12) [inline redux/reducersComp/cryptosSlice.ts]: use destructuring
({ crypto })

**Review by FedericoLeiva12 — CHANGES_REQUESTED** [review]
